loadbalancer/algorithm.go

```diff
@@ -121,10 +117,29 @@ func withFadeIn(rnd *rand.Rand, ctx *routing.LBContext, wi []int, wf []float64,
 	)
 
 	if rnd.Float64() < f {
-		return ctx.Route.LBEndpoints[choice]
+		return ep[choice]
+	}
+	for i := 0; i < len(ep); i++ {
+		if _, fadingIn := fadeInState(now, ctx.Route.LBFadeInDuration, ep[i].Detected); !fadingIn {
+			notFadingIndexes = append(notFadingIndexes, i)
+		}
 	}
 
-	return shiftToRemaining(rnd, ctx, wi, wf, now)
+	switch a := algo.(type) {
+	case *roundRobin:
+		return shiftToRemaining(a.rnd, ctx, notFadingIndexes, a.fadingWeights, now)
+	case *random:
+		return shiftToRemaining(a.rand, ctx, notFadingIndexes, a.fadingWeights, now)
+	case consistentHash:
+		// If all endpoints are fading, normal consistent hash result
+		if len(notFadingIndexes) == 0 {
```

Comment:
should not this be the default for all algorithms?

Reply:
Currently, it looks like random and roundRobin randomly select an endpoint, but weighted by how old the endpoints are so that older endpoints are more likely to be selected. For fading with consistentHash, I was thinking to just do normal consistent hash without caring how old the endpoints were

Reply:
yes makes sense